extensions/identity-hub/src/main/java/org/eclipse/dataspaceconnector/identityhub/store/IdentityHubStore.java

```diff
@@ -0,0 +1,16 @@
+package org.eclipse.dataspaceconnector.identityhub.store;
+
+import org.eclipse.dataspaceconnector.identityhub.dtos.HubObject;
+
+import java.util.Collection;
+
+/**
+ * IdentityHubStore store used to store HubObjects.
+ * A custom IdentityHubStore store can be injected via service loader as an EDC Service extension.
+ */
+public interface IdentityHubStore {
```

Comment:
Suggestion: In registration service we have divided in-memory store implementation between 3 modules. So, for consistency and keep one approach we can do the same here

reg service pr is : https://github.com/agera-edc/RegistrationServiceFork/pull/3/files#diff-cace75cb056f4c2b3bef6b4b4d508e144944e0777ae3d42153bfd9d8c8f24ba0

1. spi - which contains interface for store.
2. in-memory store.
3. actual api which usage store.

Reply:
Let's maybe revisit that in Reg Service PR though and make in-mem store a default?

Reply:
I've adapted the Reg Service PR accordingly

Reply:
I have refactored the modules to mirror what we have in the registration service:
1- spi module for store
2- main module with default in memory implementation for store